application/controllers/admin/database.php

```diff
@@ -881,129 +883,137 @@ private function actionUpdateSurveyLocaleSettings($iSurveyID)
                     if (count($data) > 0) {
                         $oSurveyLanguageSetting = SurveyLanguageSetting::model()->findByPk(array('surveyls_survey_id'=>$iSurveyID, 'surveyls_language'=>$langname));
                         $oSurveyLanguageSetting->setAttributes($data);
-                        $oSurveyLanguageSetting->save(); // save the change to database
+                        // Save the change to database
+                        if (!$oSurveyLanguageSetting->save()) { 
+                            $saveSurvey = false;
+                            Yii::app()->setFlashMessage(Chtml::errorSummary($oSurveyLanguageSetting,Chtml::tag("p",array('class'=>'strong'),gT("Survey could not be updated, please fix the following error:"))), "error");
```

Comment:
Throw exception here, and add flash message in the `catch` block instead.

Reply:
Or just throw an exception and let Yii error handler show it.

Reply:
OK, I'm getting some understanding of your fix. database.php is _not_ an easy file to work with, with so big methods. I pushed a fix where I split it into two methods instead, and added a try-catch block around those methods.

Reply:
Nice! Will take into account the advice. Agree: Sometime code reorgs are needed.

Reply:
Yeah, not an easy call as consultant. In-house I can do "bolder" changes. But you can always ask us.

Reply:
Yes, when it come to fixes,I try to keep the scope tight

Reply:
That's exactly right, as small fix as possible (with any given interpretation of "possible" ^^ ), to make cherry-picking between branches easier.